matrix_sdk/src/client.rs

```diff
@@ -867,13 +877,76 @@ impl Client {
         Ok(())
     }
 
-    /// Add `EventHandler` to `Client`.
+    /// Register a handler for a specific event type.
+    ///
+    /// The handler is a function or closue with one or more arguments. The
+    /// first argument is the event itself. All additional arguments are
+    /// "context" arguments: They have to implement [`EventHandlerContext`][].
+    /// This trait is named that way because most of the types implementing it
+    /// give additional context about an event: The room it was in, its raw form
+    /// and other similar things. As an exception to this,
+    /// [`matrix_sdk::Client`] also implements the `EventHandlerContext` trait
+    /// so you don't have to clone your client into the event handler manually.
+    ///
+    /// Invalid context arguments, for example a [`Room`][room::Room] as an
```

Comment:
How should people know what the correct context arguments are for the individual event types? Could we list those here?

Reply:
I guess I could rewrite this to only task about the `Room` context? I don't think it will apply to any others (`EncryptionInfo` only really makes sense to be extracted as `Option<EncryptionInfo>`).

Reply:
I think it'll make sense to attach a `VerificationRequest` and/or a `Verification` enum with all the `m.key.verification.*` events in the future so listing stuff explicitly out might make sense right away (under the assumption that this will be possible).